### ISSUE TITLE
Unable to delete photos

### ISSUE THREAD

**Issue by cavon08** [kollerlukas/Camera-Roll-Android-App#63]
After the message "Grant removable storage permission" and selecting the storage root folder, photos reappear

**Comment by kollerlukas** [comment]
What Android Version are you on?

**Comment by cavon08** [comment]
7.1.2

**Comment by kollerlukas** [comment]
What device are you using?

**Comment by cavon08** [comment]
Motorola Moto G3, it's also running Lineage OS 14.1

**Comment by kollerlukas** [comment]
Was this issue fixed with the last upate?

**Comment by cavon08** [comment]
No. I'm trying to delete files (screenshots, photos, videos etc.) from a SD card, maybe there's a problem because of that (?). I select the file, click the bin icon, it appears the "select root folder..." message and I select the SD card. Previously the file was deleted but reappeared afterwards, now with this last update, after selecting the SD card the "select root folder..." message pops up again and again, without the file being deleted.

**Comment by kollerlukas** [comment]
Does this #64 (using the LinageOS file manager) also fix this issue?

**Comment by cavon08** [comment]
No, I use the LineageOS File Manager regularly. I forgot to mention that I did format my SD card as internal storage when I got my phone, maybe there's a problem because of that.

**Comment by Johndeep** [comment]
Second this, also have this issue (on LineageOS 7.1.2, Samsung S4 mini, not that the device matters but formatted as internal storage also), which means that this is probably an issue about some permission stuff.

I have currently the version v1.0.3 provided by F-Droid.

**Comment by comamondo** [comment]
Same problem on Moto G4 LineageOS 14.1. Sdcard ist formatted as "normal" storage. After deleting a file the "give permission" dialog pops up for one second or so and disappears. Permissions are granted in systemsettings for apps.

**Comment by Wiredframe** [comment]
This issue persists on my Nokia 2 with an SD Card formatted as internal storage.

**Comment by floppyD** [comment]
Just bumping this issue, I just found this app and it's pretty good, but I just can't delete files. I'm on 7.1.2 and using adopted storage too

**Comment by eauchat** [comment]
I have a similar issue.
When I delete a photo, it takes a very long time (like 5-10 seconds, during which a notification is displayed that an image is being deleted). 
Sometimes the photo didn't delete. I was testing it just now, and the deletion proceeds.
But it's very slow, and the folder gallery display doesn't refresh properly after deletion: the image thumbnail is still visible in gallery and if I click on it, it'll open the next one. I have to close the folder and reopen it to have the display refreshed.

Also I'm not sure it's related, but in case that's relevant: it takes a long time (like 5 seconds after deletion is completed in camera-roll) for another gallery app to actually notice the deletion of the file.

**Comment by kollerlukas** [comment]
@eauchat Are you also trying to delete files from adopted storage?
I am sorry but at the moment I don't have a phone that takes an external Sd card. So currnently it is pretty difficult for me to fix this issue.

**Comment by eauchat** [comment]
@kollerlukas Sorry for the long delay in answering, long time without connection.
So I finally go to figure out the pattern of the behaviour. If I delete a photo in the internal storage it's pretty fast (less than one second), if in the SD card, it's very long, like 5sec per image. Which can become really problematic when you delete a bunch of images. Especially since you'll have displaying errors if you start navigating in the app while it's deleting.
I think the core deletion process is good (means it deletes really what you want it to delete). But you may see the wrong thumbnail deleted, and I scared myself a couple of time thinking I deleted the wrong image. However if you close and reopen, everything's back to normal.

**Comment by n8plus8** [comment]
If the SD card chooser doesn't appear: Enable the built-in file manager (Files 7.1.x) if you have disabled / frozen it.

**Comment by fxnn** [comment]
Same issue on my Shift5me, Android 8 (with Google Play Services). Infinite dialog loop, though I always select the SD card root. Sometimes, the apps info dialog only appears for a hundred milliseconds or so and then disappears again, without the file being deleted.

Android reports the storage permission as granted.
